File: 3x+1/go/3x+1.go
package main

import (
	"fmt"
	"time"
)

func main() {
	total_n := 1000000
	max_n := 0
	max_count := 0
	start_time := time.Now().UnixNano()

	for n := 1; n <= total_n; n++ {
		count := 0
		var i int64 = int64(n)
		for i > 1 {
			if i%2 == 0 {
				i = i / 2
			} else {
				i = 3*i + 1
			}
			count++
		}
		if count > max_count {
			max_count = count
			max_n = n
		}
	}

	end_time := time.Now().UnixNano()

	fmt.Printf("%d %d %d ms\n", max_n, max_count, (end_time-start_time)/1000000)
}

File: 3x+1/py/3x+1.py
import time
import math

#boilerplate
total_n = int(1000000)
i = 0
max_n = 0
max_count = 0
start_time = time.time()

#calculate
for i in range(total_n):
    n = i + 1
    count = 0
    while n != 1:
        if n % 2 == 0:
            n = n / 2
        else:
            n = 3 * n + 1
        count += 1
    
    if count > max_count:
        max_count = count
        max_n = i

#print results
elapsed_time = math.floor((time.time()-start_time)*1000)

print(max_n, max_count, elapsed_time, "ms")